README.md

```diff
@@ -16,6 +16,18 @@ A possible outcome would be an emulator that can run RV32I assembly.
   - follow instructions on https://nixos.wiki/wiki/Flakes#Enable_flakes
 - (Optional) install [`direnv`](https://direnv.net)
 
+# Project structure
+- The structure of the project is dictated by the tools we are using, namely Nix and Cabal. This repository is both a Nix flake and a Cabal package and as such contains the following folders/files:
+  - [`app/`](/app): Contains Haskell source for an executable that can be built and run by `cabal run`. It does not do anything meaningful at the moment.
+  - [`src/`](/src): Haskell library folder, each file inside is a Haskell module providing functions-
+  - [`test/`](/test) : Contains Haskell source for an executable that is built and run by `cabal test`, used to test the functions defined in `src/`.
+  - [`CHANGELOG.md`](CHANGELOG.md): This file will only become relevant in case of a release.
+  - [`flake.nix`](flake.nix): This is a [Nix flake](https://nixos.wiki/wiki/Flakes) providing dependencies for the project, namely GHC and Cabal.
+  - [`flake.lock`](flake.lock): File specifying the exact versions of dependencies, created by the [`nix flake lock`](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake-lock.html) command. 
+  - [`LICENSE`](LICENSE): The Apache 2.0 license.
+  - [`riscvs-hs.cabal`](riscvs-hs.cabal): Cabal [package description](https://cabal.readthedocs.io/en/3.4/cabal-package.html).
+  - [`shell.nix`](shell.nix): For compatibility with non-flake-enabled Nix versions and tools requiring .nix files, such as the [Nix environment selector](https://github.com/arrterian/nix-env-selector) plugin for VSCode.
```

Comment:
```suggestion
  - [`shell.nix`](shell.nix): For compatibility with Nix without flakes, as well as other tools.
```

The details may change. Don't think "non-flake-enabled" is an actual word.

Reply:
Let's explain how to connect VSCode with the Nix shell in a separate section, perhaps "Notes" or "Protips".